### PR TITLE
Osod fixed

### PR DESCRIPTION
**Pull request by jakesabathia2**
fix
test_get
test_export_coreml
test_create_with_single_image
test_repr

### CODE REVIEW

File: src/python/turicreate/test/test_one_shot_object_detector.py [hunk 1]
```diff
@@ -110,15 +111,17 @@ def setUpClass(self):
            'detector': lambda x: isinstance(x, tc.object_detector.object_detector.ObjectDetector),
            '_detector_version': lambda x: x==1
         }
+
+        if USE_CPP:
```
Comment (jakesabathia2) [inline src/python/turicreate/test/test_one_shot_object_detector.py]: fix test_get here

File: src/python/turicreate/toolkits/object_detector/object_detector.py [hunk 2]
```diff
@@ -269,14 +269,16 @@ def create(dataset, annotations=None, feature=None, model='darknet-yolo',
...
             'grid_width': params['grid_shape'][1],
-            'max_iterations': max_iterations,
             'mlmodel_path' : params['mlmodel_path'],
             'classes' : classes,
             'compute_final_metrics' : False
         }
         if batch_size > 0:
             tf_config['batch_size'] = batch_size
 
+        if max_iterations > 0:
```
Comment (jakesabathia2) [inline src/python/turicreate/toolkits/object_detector/object_detector.py]: fix bug for OD which is the reason of the failure of test_create_single_image

File: src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py [hunk 3]
```diff
@@ -209,7 +210,12 @@ def export_coreml(self, filename,
         >>> model.export_coreml('one_shot.mlmodel')
         """
         from coremltools.models.utils import save_spec as _save_spec
-        model = self.__proxy__['detector']._create_coreml_model(include_non_maximum_suppression=include_non_maximum_suppression, iou_threshold=iou_threshold, confidence_threshold=confidence_threshold)
+        import coremltools
+        if USE_CPP:
```
Comment (jakesabathia2) [inline src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: fixed test_export_coreml

File: src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py [hunk 4]
```diff
@@ -256,6 +262,9 @@ def __repr__(self):
         Print a string description of the model when the model name is entered
         in the terminal.
         """
+        if USE_CPP:
```
Comment (jakesabathia2) [inline src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: fix test_repr

File: src/python/turicreate/toolkits/object_detector/object_detector.py (hunk 2 above)
Comment (Jarvi-Izana) [reply src/python/turicreate/toolkits/object_detector/object_detector.py]: what do you mean?

File: src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py [hunk 5]
```diff
@@ -209,7 +210,12 @@ def export_coreml(self, filename,
         >>> model.export_coreml('one_shot.mlmodel')
         """
         from coremltools.models.utils import save_spec as _save_spec
-        model = self.__proxy__['detector']._create_coreml_model(include_non_maximum_suppression=include_non_maximum_suppression, iou_threshold=iou_threshold, confidence_threshold=confidence_threshold)
+        import coremltools
+        if USE_CPP:
+            self.__proxy__['detector'].export_coreml(filename, include_non_maximum_suppression, iou_threshold, confidence_threshold)
+            model = coremltools.models.MLModel(filename).get_spec()
+        else:
+            model = self.__proxy__['detector']._create_coreml_model(include_non_maximum_suppression=include_non_maximum_suppression, iou_threshold=iou_threshold, confidence_threshold=confidence_threshold)
```
Comment (Jarvi-Izana) [inline src/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: could we use pep8 format...

File: src/python/turicreate/toolkits/object_detector/object_detector.py (hunk 2 above)
Comment (jakesabathia2) [reply src/python/turicreate/toolkits/object_detector/object_detector.py]: the default value on the python side for `max_iteration` is `0`, which causes an assert error in c++ size which assert `max_iteration` > 0.
So the fix here didn't pass the argument if `max_iteration == 0`, so in the c++ side,
an `estimate_max_iteration()` function is called to produce `max_iterations` automatically.

Comment (Jarvi-Izana) [reply src/python/turicreate/toolkits/object_detector/object_detector.py]: good. You should leave a comment max_iteration is selected automatically if not provided.

Comment (jakesabathia2) [reply src/python/turicreate/toolkits/object_detector/object_detector.py]: done thanks!

**Comment by jakesabathia2** [comment]
Have cleaned up the formatting.
And skip OD's `test_create_with_invalid_annotations_list_coord`.

**Comment by jakesabathia2** [comment]
Gitlab internal test passes for the OD and OSOD part.

**Pull request merged**